core/src/main/scala/kafka/log/Log.scala

```diff
@@ -485,7 +485,13 @@ class Log(val dir: File,
     // but if that segment doesn't contain any messages with an offset greater than that
     // continue to read from successive segments until we get some messages or we reach the end of the log
     while(entry != null) {
-      val fetchInfo = entry.getValue.read(startOffset, maxOffset, maxLength)
+      val maxPosition = {
+        if (entry == segments.lastEntry())
+          Option(nextOffsetMetadata.relativePositionInSegment.toLong)
```

Comment:
A minor issue, but one usually uses `Some` instead of `Option` here. The latter is generally used for items that could potentially be null (and it converts those to a `None`).